### PR TITLE
Standardize comments and line separators for kpf, matlab and mvg directories.

### PR DESCRIPTION
**Pull request by gbrooksby**

### CODE REVIEW

**Comment by gbrooksby** [comment]
Jenkins test this please.

File: arrows/matlab/mxarray.h [hunk 1]
```diff
@@ -831,247 +773,208 @@ class MxArray
...
   // MxArray& operator=(const MxArray& rhs) = delete;
 
-  /*************************************************************/
-  /**             Templated mxArray importers                 **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///***           Templated mxArray importers                ***
+  ///
+  //*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:921]: This needs some manual massaging, and it doesn't look like it's *supposed* to be a Doxygen comment.

I'm not sure you want to be making this sort of change, but this looks like a candidate for a region marker (`//BEGIN` / `//END`).

File: arrows/matlab/mxarray.h [hunk 2]
```diff
@@ -831,247 +773,208 @@ class MxArray
...
 
-  /*************************************************************/
-  /**             Templated mxArray exporters                 **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///***          Templated mxArray exporters                 ***
+  ///
+  //*************************************************************
+
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:975]: Likewise. Also, the added (and thus double) blank line is against the style guidelines.

File: arrows/matlab/mxarray.h [hunk 3]
```diff
@@ -1107,68 +1008,67 @@ class MxArray
...
                                                  >::type* value );
 
-  /*************************************************************/
-  /**             Templated mxArray getters                   **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///***          Templated mxArray getters                   ***
+  ///
+  //*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1015]: Likewise.

File: arrows/matlab/mxarray.h [hunk 4]
```diff
@@ -1107,68 +1008,67 @@ class MxArray
...
                           mwIndex index, T* value );
 
-  /*************************************************************/
-  /**           Templated mxArray element setters             **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///***        Templated mxArray element setters             ***
+  ///
+  //*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1043]: Likewise.

File: arrows/matlab/mxarray.h [hunk 5]
```diff
@@ -1107,68 +1008,67 @@ class MxArray
...
                            mwIndex index, const T& value );
 
-  /*************************************************************/
-  /**    Assignment helpers (for MxArray.to<type>(value))     **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///*** Assignment helpers (for MxArray.to<type>(value))     ***
+  ///
+  //*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1069]: Likewise.

File: arrows/matlab/mxarray.h [hunk 6]
```diff
@@ -1381,12 +1271,13 @@ class MxArray
...
 
 
-  /*************************************************************/
-  /**  Assignment helpers (for MxArray.set<type>(i, value))  **/
-  /*************************************************************/
+  //*************************************************************
+  ///
+  ///*** Assignment helpers (for MxArray.set<type>(i, value)) ***
+  ///
+  //*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1278]: Likewise.

File: arrows/matlab/mxarray.h [hunk 7]
```diff
@@ -1480,20 +1370,19 @@ class MxArray
...
 };
 
-/*************************************************************/
-/**             Templated mxArray importers                 **/
-/*************************************************************/
+//*************************************************************
+///
+///***          Templated mxArray importers                 ***
+///
+//*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1383]: Likewise.

File: arrows/matlab/mxarray.h [hunk 8]
```diff
@@ -1673,12 +1559,13 @@ MxArray::fromInternal( const typename std::enable_if<
...
 
 
-/*************************************************************/
-/**             Templated mxArray exporters                 **/
-/*************************************************************/
+//*************************************************************
+///
+///***          Templated mxArray exporters                 ***
+///
+//*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1566]: Likewise.

File: arrows/matlab/mxarray.h [hunk 9]
```diff
@@ -1760,12 +1646,13 @@ MxArray::toInternal( const mxArray*                   array,
...
 
 
-/*************************************************************/
-/**             Templated mxArray getters                   **/
-/*************************************************************/
+//*************************************************************
+///
+///***          Templated mxArray getters                   ***
+///
+//*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1653]: Likewise.

File: arrows/matlab/mxarray.h [hunk 10]
```diff
@@ -1864,12 +1751,13 @@ MxArray::atInternal( const mxArray*     array,
...
 
 
-/*************************************************************/
-/**             Templated mxArray setters                   **/
-/*************************************************************/
+//*************************************************************
+///
+///***          Templated mxArray setters                   ***
+///
+//*************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxarray.h:1758]: Likewise.

File: arrows/matlab/mxtypes.h [hunk 11]
```diff
@@ -39,14 +38,16 @@ namespace kwiver {
...
-   Don't use with function templates due to type promotion!
-   (Ignore it and get PITA!)                                */
-/************************************************************/
+//************************************************************
+///
+/// Traits for fundamental datatypes.
+/// Don't use with function templates due to type promotion!
+/// (Ignore it and get PITA!)
+///
+//************************************************************
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxtypes.h:47]: Again, here, this probably should be manually adjusted. It's also *actually* not a Doxygen comment.

File: arrows/matlab/mxtypes.h [hunk 12]
```diff
@@ -85,26 +84,27 @@ struct MxIntTy<T, typename std::enable_if<
...
 
-/**********************************************/
-/* Introducing traits for MATLAB array types. */
-/**********************************************/
+//**********************************************
+///
+/// Introducing traits for MATLAB array types.
+///
+//**********************************************
+
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxtypes.h:100]: Likewise. (And again, an extra blank line.)

File: arrows/matlab/mxtypes.h [hunk 13]
```diff
@@ -249,48 +249,46 @@ struct MxTypes<T, typename std::enable_if<
...
 
-/********************************************/
-/* Type traits for function template usage. */
-/********************************************/
+//********************************************
+///
+/// Type traits for function template usage.
+///
+//********************************************
+
```
Comment (mwoehlke-kitware) [inline arrows/matlab/mxtypes.h:257]: Likewise.

File: arrows/mvg/sfm_utils.h [hunk 14]
```diff
@@ -226,22 +214,18 @@ clean_cameras_and_landmarks(
...
- * The alignment in this case means that the dot product of these vectors
- * is postive, not that the vectors are equal.
- * Up in the image is the negative Y-axis.  Up in the world defaults to
- * the postive Z-axis, but this is configurable by specifying \p up.
- */
+//  Test if "up" in the image aligns with "up" in the world for each camera.
+//  The alignment in this case means that the dot product of these vectors
+//  is postive, not that the vectors are equal.
+//  Up in the image is the negative Y-axis.  Up in the world defaults to
+//  the postive Z-axis, but this is configurable by specifying \p up.
```
Comment (mwoehlke-kitware) [inline arrows/mvg/sfm_utils.h:221]: It looks like this was intended to be a Doxygen comment.

File: arrows/mvg/sfm_utils.h [hunk 15]
```diff
@@ -226,22 +214,18 @@ clean_cameras_and_landmarks(
...
 KWIVER_ALGO_MVG_EXPORT
 bool
 camera_upright(vital::camera_perspective const& camera,
                vital::vector_3d const& up = vital::vector_3d(0, 0, 1));
 
 /// Return true if most cameras are upright
-/*
- * \sa camera_upright
- */
+//  \sa camera_upright
```
Comment (mwoehlke-kitware) [inline arrows/mvg/sfm_utils.h:228]: It looks like this was intended to be a Doxygen comment.

**Review by mwoehlke-kitware — CHANGES_REQUESTED** [review]
Given how large and somewhat messy they are, I'd be tempted to drop the matlab files from this...

File: arrows/kpf/detected_object_set_output_kpf.cxx [hunk 16]
```diff
@@ -129,29 +129,27 @@ write_set( const kwiver::vital::detected_object_set_sptr set,
...
-             << bbox.max_y() << " " // 13: BR-y
-             << bbox.area() << " "  // 14: area
-             << "0 "                // 15: world-loc x
-             << "0 "                // 16: world-loc y
-             << "0 "                // 17: world-loc z
-             << "-1 "                // 18: timestamp
-             << (*det)->confidence()   // 19: confidence
-             << std::endl;
-    */
+    // This is how kw18 writes out data, Delete when we are happy with the above code
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply arrows/kpf/detected_object_set_output_kpf.cxx:132]: I've put this back to the original form. I agree it makes it easier to put back if desired.

File: arrows/kpf/vital_kpf_adapters.h [hunk 17]
```diff
@@ -47,26 +45,25 @@ struct vital_box_adapter_t : public KPF::kpf_box_adapter< kwiver::vital::boundin
...
-      // should check that d's vectors are the same length
-      for (size_t i = 0; i<d.poly_x.size(); ++i) {
-        p.xy.push_back(make_pair(d.poly_x[i], d.poly_y[i]));
-      }
-      return p; })
-  {}
-};
-*/
\ No newline at end of file
+// I do not think vital has this data associated with detected objects...
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply arrows/kpf/vital_kpf_adapters.h:48]: Fixed.

File: arrows/matlab/matlab_detection_output.cxx [hunk 18]
```diff
@@ -105,7 +103,8 @@ class matlab_detection_output::priv
     eval( "detector_initialize()" );
   }
 
-  // --- instance data -----
+  // --------------------------------------------------------------------------
+ instance data -----
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply arrows/matlab/matlab_detection_output.cxx:107]: Fixed.

File: arrows/matlab/matlab_image_object_detector.cxx [hunk 19]
```diff
@@ -150,7 +144,8 @@ class matlab_image_object_detector::priv
     eval( "detector_initialize()" );
   }
 
-  // --- instance data -----
+  // --------------------------------------------------------------------------
+ instance data -----
```
[the remark this answers is not in the document]
Comment (gbrooksby) [reply arrows/matlab/matlab_image_object_detector.cxx:148]: Fixed.

File: arrows/matlab/mxtypes.h (hunk 11 above)
Comment (gbrooksby) [reply arrows/matlab/mxtypes.h:47]: Fixed.

File: arrows/matlab/mxtypes.h (hunk 12 above)
Comment (gbrooksby) [reply arrows/matlab/mxtypes.h:100]: Fixed.

File: arrows/mvg/sfm_utils.h (hunk 14 above)
Comment (gbrooksby) [reply arrows/mvg/sfm_utils.h]: Fixed.  The doxygen is corrected now.

File: arrows/mvg/sfm_utils.h (hunk 15 above)
Comment (gbrooksby) [reply arrows/mvg/sfm_utils.h]: Fixed.  The doxygen is corrected now.

File: arrows/matlab/matlab_image_filter.cxx [hunk 20]
```diff
@@ -143,7 +137,8 @@ class matlab_image_filter::priv
     eval( "filter_initialize()" );
   }
 
-  // ------- instance data -------
+  // --------------------------------------------------------------------------
```
Comment (mleotta) [inline arrows/matlab/matlab_image_filter.cxx:140]: This looks problematic

File: arrows/matlab/mxarray.h [hunk 21]
```diff
@@ -1,79 +1,76 @@
-/*
-
-Copyright (c) 2014 Kota Yamaguchi
```
Comment (mleotta) [inline arrows/matlab/mxarray.h:3]: This looks like third party code, we probably shouldn't reformat this file.

File: arrows/matlab/mxtypes.h [hunk 22]
```diff
@@ -1,32 +1,31 @@
-/** MxTypes and other type traits for template.
-
-Copyright (c) 2014 Kota Yamaguchi
```
Comment (mleotta) [inline arrows/matlab/mxtypes.h:3]: We probably should also not reformat this file since it came from a third party source

**Review by mleotta — CHANGES_REQUESTED** [review]
This mostly looks good, but I found one case that looks broken and a couple of files in matlab that look like they came from third party sources and should probably not be reformatted.  If we can revert these few things this will be good to go.

**Pull request merged**